lib/util/ldaputil.js

```diff
@@ -76,29 +76,135 @@ export const login = function(options, username, password) {
         })
         return resolver.promise
     }
+    function findUserGroups(client, userDN) {
+        const resolver = Promise.defer()
```

Comment:
Do not use Promise.defer. Use proper ES6 Promises

See https://nodejs.org/api/util.html#utilpromisifyoriginal if needed